scripts/switch-to-preproduction.sh

```diff
@@ -0,0 +1,8 @@
+#!/bin/sh
+
+[ -z "$PREPRODUCTION_PIPELINE" ] || {
+  echo "Switch to preproduction package.json"
+  cp -f .azure-pipelines-preproduction/package.json ./package.json
+  cp -f .azure-pipelines-preproduction/package-lock.json ./package-lock.json
+  PREPRODUCTION_PIPELINE= npm install
```

Comment:
What's the meaning of this assignment expression? Does this shell make sure to executes after the production pipeline?

Reply:
To make sure that second run will not cause any potential infinity loop. This assignment will clear the PREPRODUCTION_PIPELINE

Reply:
So PREPRODUCTION_PIPELINE is more like a state variable. If "npm install" command here used by other places, otherwise "npm install" will cause some misunderstanding.